src/main/java/seedu/address/logic/parser/IndexHandler.java

```diff
@@ -35,9 +36,41 @@ public ContactIndex assignIndex() {
         return new ContactIndex(availableIndex);
     }
 
+    /**
+     * Gets the first person who satisfies the index, if any.
+     */
     public Optional<Person> getPersonByIndex(ContactIndex index) {
         List<Person> personList = model.getObservablePersonList();
         return personList.stream().filter(person -> person.getContactIndex().equals(index)).findFirst();
     }
 
+    /**
+     * Assigns a contact index for a Recommendation.
+     * @return new Contact Index for new Recommendation.
+     */
+    public ContactIndex assignRecommendationIndex() {
+        List<Recommendation> personList = model.getObservableRecommendationList();
+        if (personList.isEmpty()) {
+            return new ContactIndex(1);
+        }
+        OptionalInt takenIndices = IntStream.iterate(1, integer -> personList.stream()
+                        .anyMatch(recommendation -> recommendation
+                                .getContactIndex().equals(new ContactIndex(integer))), x -> x + 1).max();
+
+        assert takenIndices.isPresent();
+
+        int availableIndex = takenIndices.getAsInt() + 1;
+        return new ContactIndex(availableIndex);
```

Comment:
Try to adhere to DRY. It is similar to `assignIndex()` no?

Reply:
Yes it is. It's your stuff so I didn't want to change it.

Reply:
It uses a different list to get a contact index, so I can't simply use `assignIndex()`.